code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceConsumeMapDaoImpl.java

```diff
@@ -46,4 +46,13 @@ public ServiceConsumeMap findNonRemovedMap(long serviceId, long consumedServiceI
                                 .and(SERVICE_CONSUME_MAP.REMOVED.isNull())).fetchInto(ServiceConsumeMapRecord.class);
     }
 
+    @Override
+    public List<? extends ServiceConsumeMap> findMapsToRemove(long serviceId) {
```

Comment:
Oops, you are right. Forgot to add one more condition here - or map.state='removing'.